### PR TITLE
feat: [Option] add data attribute to OptionProps type definition

### PR DESCRIPTION
**Pull request by t49tran**
Add `data` attribute to `OptionProps` type defition of `Option` component.

The issue raised in #3038

### CODE REVIEW

File: src/components/Option.js [hunk 1]
```diff
@@ -33,6 +33,8 @@ export type OptionProps = PropsWithStyles &
     /* Type is used by the menu to determine whether this is an option or a group.
     In the case of option this is always `option`. */
     type: 'option',
+    /* The data of the option rendered in the Option component */
+    data: any
```
Comment (pmusial9) [inline src/components/Option.js]: Do you know why that generic type has been removed? It'd be nice to have that type here 🙂 

image

**Comment by carl-coolblue** [comment]
We're needing this too 👍

**Comment by IanVS** [comment]
It looks like this has already been merged in https://github.com/JedWatson/react-select/pull/3154.

**Comment by IanVS** [comment]
@t49tran would you mind closing this PR, to help whittle down the number of open PRs in the project?

**Pull request closed without merging**